app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/StripePaymentActivity.java

```diff
@@ -276,21 +296,31 @@ private String buildCardString(@NonNull SourceCardData data) {
         return data.getBrand() + ",  " + getString(R.string.creditcard_ending_in) + "  " + data.getLast4();
     }
 
-    private int reserveTicket() {
+    private void reserveTicket() {
         int ticketType = getIntent().getIntExtra("ticketType", -1);
-        int memberID = 0; //TODO: get real user ID (id used in database)
-
-        if (ticketType < 0) {
+        if (ticketType <= 0) {
             Toast.makeText(getApplicationContext(), R.string.internal_error, Toast.LENGTH_LONG).show();
             finish();
         }
 
+        //progressDialog = ProgressDialog.show(StripePaymentActivity.this, "", getString(R.string.purchase_progress_message), true);
```

Comment:
`// TODO`?

Reply:
I worked that out.

Also, I moved the ticket reservation to the BuyTicketActivity (the ticket summary), since we want to reserve a ticket at the moment the user presses the purchase button, regardless of what the following activity will be (Stripe or Payment Overview)